src/pages/ModuleEditor/ModuleEditor.tsx

```diff
@@ -0,0 +1,76 @@
+import * as React from 'react';
+import {useState} from 'react';
+import {Form, Col} from 'react-bootstrap';
+import styles from '../ResetPassword/ResetPassword.module.scss';
+import {Layout} from '../Layout/Layout';
+import {Formik, FormikHelpers} from 'formik';
+import Input from '../../components/util/Input/Input';
+import {ModuleEditorForm, ModuleEditorSchema} from './ModuleEditorSchema';
+import {LoadingButton} from '../../util/LoadingButton';
+import {handleAxiosError, 
+  isBadRequest, makeMessageState} from '../../util';
+import {Message} from '../../util/Message';
+// import {submitContactRequest} from '../../api';
+import { v4 as uuidv4 } from 'uuid';
+
+const initialValues: ModuleEditorForm = {moduleName: '', shareLink: '', description: ''};
+function generateLink(uuid:String)
```

Comment:
We only store the UUID so there is no need to include the whole URL. Base this form off of the module interface. This field is called 'specialCode'